<?xml version="1.0"?>

<launch>
	
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>

  <param name="/use_sim_time" value="true"/>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find overtaking_maneuver)/gazebo/overtaking.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
  

   <node name="path_creator" pkg="overtaking_maneuver" type="path_creator.py">
		<param name="use_sim_time" 		value="true" />
	</node> 

  <node name="control" pkg="overtaking_maneuver" type="control.py">
    <param name="path_topic" 		value="/path" />
    <param name="odom_topic_name" 		value="/odom" />
    <param name="cmd_topic_name" 		value="/cmd_vel" />
		<param name="use_sim_time" 		value="true" />
  </node>

  <node name="scan" pkg="overtaking_maneuver" type="scan.py">
		<param name="use_sim_time" 		value="true" />
	</node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find overtaking_maneuver)/rviz/Rviz.rviz"/>


</launch>

<!-- roslaunch turtlebot3_gazebo turtlebot3_empty_world.launch-->
<!-- roslaunch turtlebot3_teleop turtlebot3_teleop_key.launch -->
<!-- roslaunch turtlebot3_slam turtlebot3_slam.launch -->
